packages/datadog-plugin-aws-sdk/test/sqs.spec.js

```diff
@@ -268,6 +296,176 @@ describe('Plugin', () => {
+          tracer = require('../../dd-trace')
+          tracer.use('aws-sdk', { sqs: { dsmEnabled: true } })
+        })
+
+        before(async () => {
+          return agent.load('aws-sdk', {
+            sqs: {
+              consumer: false,
+              dsmEnabled: true
+            }
+          },
+          { dsmEnabled: true })
+        })
+
+        before(done => {
+          AWS = require(`../../../versions/${sqsClientName}@${version}`).get()
+
+          sqs = new AWS.SQS({ endpoint: 'http://127.0.0.1:4566', region: 'us-east-1' })
+          sqs.createQueue(queueOptions, (err, res) => {
+            if (err) return done(err)
+
+            done()
+          })
+        })
+
+        after(done => {
+          sqs.deleteQueue({ QueueUrl: QueueUrl }, done)
+        })
+
+        after(() => {
+          return agent.close({ ritmReset: false })
+        })
+
+        it('Should set pathway hash tag on a span when producing', (done) => {
+          sqs.sendMessage({
+            MessageBody: 'test DSM',
+            QueueUrl
+          }, (err) => {
+            if (err) return done(err)
+
+            let produceSpanMeta = {}
+            agent.use(traces => {
+              const span = traces[0][0]
+
+              if (span.resource.startsWith('sendMessage')) {
+                produceSpanMeta = span.meta
+              }
+
+              expect(produceSpanMeta).to.include({
+                'pathway.hash': expectedProducerHash.readBigUInt64BE(0).toString()
+              })
+            }).then(done, done)
+          })
+        })
+
+        it('Should set pathway hash tag on a span when consuming', (done) => {
+          sqs.sendMessage({
+            MessageBody: 'test DSM',
+            QueueUrl
+          }, (err) => {
+            if (err) return done(err)
+
+            sqs.receiveMessage({
+              QueueUrl,
+              MessageAttributeNames: ['.*']
+            }, (err) => {
+              if (err) return done(err)
+
+              let consumeSpanMeta = {}
+              agent.use(traces => {
+                const span = traces[0][0]
+
+                if (span.name === 'aws.response') {
+                  consumeSpanMeta = span.meta
+                }
+
+                expect(consumeSpanMeta).to.include({
+                  'pathway.hash': expectedConsumerHash.readBigUInt64BE(0).toString()
+                })
+              }).then(done, done)
+            })
+          })
+        })
+
+        describe('emits a new DSM Stats to the agent when DSM is enabled', () => {
+          before(done => {
+            sqs.sendMessage({
+              MessageBody: 'test DSM',
+              QueueUrl
+            }, (err) => {
+              if (err) return done(err)
+
+              sqs.receiveMessage({
+                QueueUrl,
+                MessageAttributeNames: ['.*']
+              }, (err) => {
+                if (err) return done(err)
+
+                tracer._tracer._dataStreamsProcessor.onInterval()
```

Comment:
Could the flushing be triggered in some other way? If this is needed, it needs to be accessible elsewhere anyway in order to allow flushing before the process exits.

Reply:
AFAIK, we flush the tracer agent exporter using `beforeExit`. I can add this beforeExit logic to flush the DSM writer, but is there a way to exit the tracer in the tests while still keeping the mock agent running? Or is there another way we handle flushing the agent for tests?

Reply:
In the short term, I'd say to either add DSM to `tracer.flush()` and call it explicitly in tests, or add an option similar to `flushInterval` for tracing. Anything that doesn't use the internals in a test.

Reply:
AFAIK we do not have a `tracer.flush()` function. The only `flush` functions I see are for the `AgentExporter` class [here](https://github.com/DataDog/dd-trace-js/blob/9b3568e1574d6dc85c6e357c2266226914b55310/packages/dd-trace/src/exporters/agent/index.js#L60). I have updated the DataStreamsProcessor class to flush upon process exit [here](https://github.com/DataDog/dd-trace-js/pull/3864/files#diff-781317475a071e97f967958ee0faa810c6d21b367cef17d0a311696aecb1125dR196:~:text=process.once(%27beforeExit%27%2C%20()%20%3D%3E%20this.onInterval())), similar to the AgentExporter. However, since we create the processor as an internal class (`tracer._dataStreamsProcessor`), this will still require calling an "internal" class.

Reply:
I will also add the `flushInterval` to the DataStreamsProcessor class 👍

Reply:
Let's just add a `flush` method on the test agent then? It's something that is needed for various things in tests, and the less we need to work with internals directly the better.

Reply:
I eliminated all internals in the test and used `flushInterval=0` to flush all pipeline stats immediately. That fixed all my errors and tests are now passing without referencing any internals. Resolving this.